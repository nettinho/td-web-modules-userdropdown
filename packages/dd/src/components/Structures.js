import _ from "lodash/fp";
import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Header, Icon, Table, Segment } from "semantic-ui-react";
import { FormattedMessage } from "react-intl";
import StructuresPagination from "./StructuresPagination";
import StructureRow from "./StructureRow";
import StructuresSearch from "./StructuresSearch";
import StructureSelectedFilters from "./StructureSelectedFilters";
import StructuresLabelResults from "./StructuresLabelResults";

export const StructuresHeader = () => (
  <Header as="h2">
    <Icon circular name="list" />
    <Header.Content>
      <FormattedMessage id="structures.header" />
      <Header.Subheader>
        <FormattedMessage id="structures.subheader" />
      </Header.Subheader>
    </Header.Content>
  </Header>
);

export const Structures = ({ structures }) => (
  <React.Fragment>
    <Segment>
      <StructuresHeader />
      <StructuresSearch />
      <StructureSelectedFilters />
      <StructuresLabelResults />
      <Table celled striped compact selectable>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell
              content={<FormattedMessage id="structure.name" />}
            />
            <Table.HeaderCell
              content={<FormattedMessage id="structure.ou" />}
            />
            <Table.HeaderCell
              content={<FormattedMessage id="structure.system" />}
            />
            <Table.HeaderCell
              content={<FormattedMessage id="structure.group" />}
            />
            <Table.HeaderCell
              content={<FormattedMessage id="structure.description" />}
            />
          </Table.Row>
        </Table.Header>
        <Table.Body>
          {structures.map(
            (s, i) => !_.isNil(s.ou) && <StructureRow key={i} {...s} />
          )}
        </Table.Body>
      </Table>
      <StructuresPagination />
    </Segment>
  </React.Fragment>
);

Structures.propTypes = {
  structures: PropTypes.array
};

const mapStateToProps = ({ structures }) => ({
  structures
});

export default connect(mapStateToProps)(Structures);
