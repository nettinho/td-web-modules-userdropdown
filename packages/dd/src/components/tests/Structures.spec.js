import React from "react";
import { shallow } from "enzyme";
import { Structures } from "../Structures";

describe("<Structures />", () => {
  it("matches the latest snapshot", () => {
    const structures = [
      {
        id: 1,
        ou: "org",
        name: "s1",
        description: "dd",
        group: "gp",
        system: "sys"
      }
    ];
    const props = { structures };
    const wrapper = shallow(<Structures {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
