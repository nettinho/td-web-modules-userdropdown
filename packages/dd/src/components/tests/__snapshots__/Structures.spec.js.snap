// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Structures /> matches the latest snapshot 1`] = `
<React.Fragment>
  <Segment>
    <StructuresHeader />
    <InjectIntl(Connect(StructuresSearch)) />
    <InjectIntl(Connect(StructureSelectedFilters)) />
    <InjectIntl(Connect(StructuresLabelResults)) />
    <Table
      as="table"
      celled={true}
      compact={true}
      selectable={true}
      striped={true}
    >
      <TableHeader
        as="thead"
      >
        <TableRow
          as="tr"
          cellAs="td"
        >
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="structure.name"
                values={Object {}}
              />
            }
          />
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="structure.ou"
                values={Object {}}
              />
            }
          />
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="structure.system"
                values={Object {}}
              />
            }
          />
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="structure.group"
                values={Object {}}
              />
            }
          />
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="structure.description"
                values={Object {}}
              />
            }
          />
        </TableRow>
      </TableHeader>
      <TableBody
        as="tbody"
      >
        <withRouter(StructureRow)
          description="dd"
          group="gp"
          id={1}
          key="0"
          name="s1"
          ou="org"
          system="sys"
        />
      </TableBody>
    </Table>
    <Connect(StructuresPagination) />
  </Segment>
</React.Fragment>
`;
